perceval/backends/core/git.py

```diff
@@ -764,25 +764,52 @@ def _handle_trailer(self, line):
     def _handle_action_data(self, data):
         modes = self.__parse_data_list(data['modes'])
         indexes = self.__parse_data_list(data['indexes'])
-        filename = data['file']
+        filename = data['newfile'] if data['newfile'] else data['file']
 
         if filename not in self.commit_files:
             self.commit_files[filename] = {}
 
         self.commit_files[filename]['modes'] = modes
         self.commit_files[filename]['indexes'] = indexes
         self.commit_files[filename]['action'] = data['action']
-        self.commit_files[filename]['file'] = filename
+        self.commit_files[filename]['file'] = data['file']
         self.commit_files[filename]['newfile'] = data['newfile']
 
+    def _handle_pending_files(self):
+        for _, data in self.pending_files.items():
+            if data['file'] in self.commit_files:
```

Comment:
Due to `data['file']` is used several times, I'd assign this value to a variable for later usage. For small repositories it won't be a problem, but for a large one, it's time that keeps adding up.

Maybe we can even speed it up storing `self.commit_files[filename]` too.